Сетка страниц и крупных блоков на флексах

comeunne opened the pull request:
на 56 строке span не является дочерним элементом для ul, но я пока не знаю каким тегом его заменить

---
:mortar_board: [Сетка страниц и крупных блоков на флексах](https://up.htmlacademy.ru/htmlcss/33/user/1755265/tasks/7)

comeunne committed:
Сетка страниц и крупных блоков на флексах

Muhammadabumansur commented:
![image](https://user-images.githubusercontent.com/16342548/155660262-5b57fa3f-d29b-460b-a496-3a42cdac03c5.png)

Это же криво. Проверь ширину у элементов. И убери фиксированную высоту. Опять ты ее ставишь везде

Muhammadabumansur commented on index.html:
```diff
@@ -32,88 +32,95 @@ <h2 class="visually-hidden">Лайк</h2>
+                <p class="advantage-description">Седона - не аттракцион для туристов, там течёт своя жизнь</p>
+              </div>
+              <img class="aside-image" src="images/city-sedona.jpg" alt="Фото города Седона" width="800" height="384">
+              <ul>
+                <span class="tips-row">
+                  <li class="tips-item">
+                    <h2 class="visually-hidden">Иконка-жильё</h2>
+                    <h3 class="tips-title tips-title-1">Жильё</h3>
+                    <p class="tips-description tips-description-1">Рекомендуем пожить, в настоящем мотеле, всё как в
+                      кино!
+                    </p>
+                  </li>
+                  <li class="tips-item">
+                    <h2 class="visually-hidden">Иконка-гамбургер</h2>
+                    <h3 class="tips-title tips-title-2">Еда</h3>
+                    <p class="tips-description tips-description-2">Всегда заказывайте топовый фирменный бургер, вы не
+                      разочаруетесь!
+                    </p>
+                  </li>
+                  <li class="tips-item">
+                    <h2 class="visually-hidden">Иконка-подарок</h2>
+                    <h3 class="tips-title tips-title-3">Сувениры</h3>
+                    <p class="tips-description tips-description-3">Не только китайского, но и настоящего местного
+                      производства!
+                    </p>
+                  </li>
+                </span>
+              </ul>
```
Удали этот span

Muhammadabumansur commented:
![image](https://user-images.githubusercontent.com/16342548/155660491-8c487493-0057-4de8-8488-df6ef6ceab62.png)

Видишь - сюда телефон как попал? Потому что ты .main-index задал фиксированную высоту

Muhammadabumansur commented:
![image](https://user-images.githubusercontent.com/16342548/155660650-39f0c5aa-cdc1-4308-ad88-fc0d2d64d162.png)

А что это за тень? Наверное ты ее хотел задать для page-container

Muhammadabumansur commented:
![image](https://user-images.githubusercontent.com/16342548/155660740-0a8d3211-095e-41d0-b89b-ae252ff269a2.png)

Тут не должно быть ни высоты ни ширины

Muhammadabumansur commented:
.page-container задай ширину и выровняй его по центру, остальным секциям ширину задавать не нужно, они будут ограничены page-container

Muhammadabumansur commented:
![image](https://user-images.githubusercontent.com/16342548/155660906-effd3cdb-4796-49cd-b696-25eba439db8b.png)

Зачем ты заголовку задаешь ширину и высоту. Ты смотри на результат, так не должно выглядеть

comeunne committed:
Убирает лишнюю ширину, телефон передвигает в конец

Muhammadabumansur commented:
Что-то не то, то ли я не то название класса написал. Тебе нужно page-wrapper задать ширину 1200px и выровнять по центру с помощью margin-left: auto и margin-right: auto; А page-container вообще надо удалить, в нем нет смысла

body поменяй min-height: 100%; на height: 100%;

page-wrapper  задай min-height: 100%; display: flex; flex-direction: column;

футеру задай margin-top: auto;

У футера падинги не обнуляй, вообще странная привычка везде обнулять падинги и марджины. Не пиши лишние стили. Их надо удалить Как раз, за счет падингов ты и будешь устанавливать правильную высоту блокам потом

Марджины это отступы вне элемента а падинги это внутренние отступы, которые внутри элемента. Они для того, чтоб контент не прилипал к краям. .sorting-filter .sorting-filter-results .pagination-container .mailing-catalog там везде ты зря используешь марджины.

Muhammadabumansur commented on styles/styles.css:
```diff
@@ -318,7 +380,17 @@ body {
+  justify-content: space-between;
+  flex-wrap: wrap;
+  list-style: none;
+  margin: 0;
+  margin:0 ;
```
Здесь даже два раза подряд обнулил марджины)

comeunne committed:
Убирает

лишние padding, margin; page-wrapper->min-height: 100%; display: flex; flex-direction: column, width 1200px, margin-left & margin-right: auto; footer-contacts: margin-top:auto; body: height: 100%; page-container удален
